base/broadcast.jl

```diff
@@ -279,8 +305,16 @@ ziptype(A) = typestuple(A)
 ziptype(A, B) = (Base.@_pure_meta; Iterators.Zip2{typestuple(A), typestuple(B)})
 @inline ziptype(A, B, C, D...) = Iterators.Zip{typestuple(A), ziptype(B, C, D...)}
 
-_broadcast_type(f, T::Type, As...) = Base._return_type(f, typestuple(T, As...))
-_broadcast_type(f, A, Bs...) = Base._default_eltype(Base.Generator{ziptype(A, Bs...), ftype(f, A, Bs...)})
+scalarify_nullable(x) = x
+scalarify_nullable(x::Nullable) = (x,)
```

Comment:
It's not quite so simple because nullables need to be treated as scalars for broadcasting with arrays, but when broadcasting nullables themselves they are treated as containers.

Do you think the approach [here](https://github.com/JuliaLang/julia/pull/16961/commits/bed109e829003f5ffba86ec64959b69c9ab38be6) is more general and less ugly?